internal/provider/metric_export_prometheus_config_resource_test.go

```diff
@@ -0,0 +1,213 @@
+		Times(3)
+
+	// Update
+	s.EXPECT().GetCluster(gomock.Any(), clusterID).
+		Return(cluster, nil, nil).
+		Times(2)
+	s.EXPECT().GetPrometheusMetricExportInfo(gomock.Any(), clusterID).
+		Return(createdPrometheusClusterInfo, nil, nil)
+	// It should not invoke EnablePrometheusMetricExport as request contains clusterID
+	// which will not get changed.
+	s.EXPECT().EnablePrometheusMetricExport(gomock.Any(), clusterID, gomock.Any()).
+		Return(updatedPrometheusClusterInfo, nil, nil).Times(0)
+	s.EXPECT().GetPrometheusMetricExportInfo(gomock.Any(), clusterID).
+		Return(updatedPrometheusClusterInfo, nil, nil).
+		Times(3)
+
+	// Delete
+	s.EXPECT().DeleteCluster(gomock.Any(), clusterID)
+	s.EXPECT().DeletePrometheusMetricExport(gomock.Any(), clusterID)
+
+	testMetricExportPrometheusConfigResource(t, clusterName, true)
+}
+
+func testMetricExportPrometheusConfigResource(t *testing.T, clusterName string, useMock bool) {
+	var (
+		clusterResourceName                      = "cockroach_cluster.test"
+		metricExportPrometheusConfigResourceName = "cockroach_metric_export_prometheus_config.test"
+	)
+
+	resource.Test(t, resource.TestCase{
+		IsUnitTest:               useMock,
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: getTestMetricExportPrometheusConfigResourceCreateConfig(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					testMetricExportPrometheusConfigExists(metricExportPrometheusConfigResourceName, clusterResourceName),
+					resource.TestCheckResourceAttr(metricExportPrometheusConfigResourceName, "status", "ENABLED"),
+				),
+			},
+			{
+				Config: getTestMetricExportPrometheusConfigResourceUpdateConfig(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					testMetricExportPrometheusConfigExists(metricExportPrometheusConfigResourceName, clusterResourceName),
+					resource.TestCheckResourceAttr(metricExportPrometheusConfigResourceName, "status", "ENABLED"),
+				),
+			},
+			{
+				ResourceName:      metricExportPrometheusConfigResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testMetricExportPrometheusConfigExists(
+	resourceName, clusterResourceName string,
+) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		p := testAccProvider.(*provider)
+		p.service = NewService(cl)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+		clusterRs, ok := s.RootModule().Resources[clusterResourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", clusterResourceName)
+		}
+
+		clusterID := clusterRs.Primary.Attributes["id"]
+		log.Printf("[DEBUG] clusterID: %s, name %s", clusterRs.Primary.Attributes["id"], clusterRs.Primary.Attributes["name"])
+
+		_, _, err := p.service.GetPrometheusMetricExportInfo(context.TODO(), clusterID)
+		if err == nil {
+			return nil
+		}
```

Comment:
Is this enough to show that its there?  I thought it could return NOT_DEPLOYED or even DEPLOYING, etc...

Reply:
We have added a check to make sure that the status is `ENABLED` in API response as well.